modules/percolator/src/main/java/org/elasticsearch/percolator/ExtractQueryTermsService.java

```diff
@@ -153,33 +178,45 @@ public static void extractQueryTerms(Query query, ParseContext.Document document
                         continue;
                     }
 
-                    Set<Term> temp;
+                    Result temp;
                     try {
                         temp = extractQueryTerms(clause.getQuery());
                     } catch (UnsupportedQueryException e) {
                         uqe = e;
                         continue;
                     }
-                    bestClause = selectTermListWithTheLongestShortestTerm(temp, bestClause);
+                    bestClause = selectTermListWithTheLongestShortestTerm(temp.terms, bestClause);
                 }
                 if (bestClause != null) {
-                    return bestClause;
+                    // 1 required clause, no prohibited clause, maybe some optional clauses, so it is verified:
+                    boolean verified = mnsm <= 1 && numProhibitedClauses == 0 && numRequiredClauses == 1;
```

Comment:
should it be `mnsm == 0` rather than  `mnsm <= 1`? if there is one MUST and one should clause and msm == 1 then this is actually a conjunction?